modular_skyrat/master_files/code/modules/events/spacevine.dm

```diff
@@ -534,6 +545,32 @@
 		space_turf.ChangeTurf(/turf/open/floor/plating/kudzu)
 		space_turf.color = hue
 
+//Hitting/crossing has a chance to infect you with a disease
+/datum/spacevine_mutation/disease_carrying
+	name = "Disease carrying"
+	hue = "#dcf597"
+	severity = 5
+	quality = NEGATIVE
+
+/datum/spacevine_mutation/disease_carrying/on_hit(obj/structure/spacevine/vine_object, mob/hitter, obj/item/item_used, expected_damage)
+	if(isliving(hitter))
+		var/mob/living/living_hitter = hitter
+		if(isvineimmune(living_hitter))
+			return
+		if(prob(20))
+			var/datum/disease/new_disease = new /datum/disease/advance/random(5, 5)
+			living_hitter.ForceContractDisease(new_disease, FALSE, TRUE)
+	. = expected_damage
+
+/datum/spacevine_mutation/disease_carrying/on_cross(obj/structure/spacevine/vine_object, mob/crosser)
+	if(isliving(crosser))
+		var/mob/living/living_crosser = crosser
+		if(isvineimmune(living_crosser))
+			return
+		if(prob(20))
+			var/datum/disease/new_disease = new /datum/disease/advance/random(5, 5)
+			living_crosser.ForceContractDisease(new_disease, FALSE, TRUE)
```

Comment:
ah yes today I will hit the vine and catch a virus for doing what I'm supposed to do.

I'm not a fan of this one and I feel like this could be problematic via disease overriding, and because I'm not sure if this respects disease-immune species.